### PR TITLE
Ref argument

### PR DESCRIPTION
**Pull request by benwiley4000**
This is the same as #1 but adds test/rubocop fixes and rebases against the latest upstream master.

### CODE REVIEW

File: lib/licensee/projects/github_project.rb [hunk 1]
```diff
@@ -39,8 +41,14 @@ def files
...
       def load_file(file)
-        client.contents(@repo, path:   file[:path],
-                               accept: 'application/vnd.github.v3.raw').to_s
+        pathname = file[:path]
+        accept = 'application/vnd.github.v3.raw'
+        ref = @revision
+        if @revision
+          client.contents(@repo, path: pathname, accept: accept, ref: ref).to_s
+        else
+          client.contents(@repo, path: pathname, accept: accept).to_s
```
Comment (francoisjacques) [inline lib/licensee/projects/github_project.rb:50]: can it be simplified with 

`client.contents(@repo, path: pathname, accept: accept, ref: ref).to_s`

and let the receiver deal with the fact that ref can be nil?

Comment (benwiley4000) [reply lib/licensee/projects/github_project.rb:50]: I did that previously, but Octokit tries to stick `?ref` on the end of the github fetch URL if I do that, which causes problems in the tests (and possibly with the actual API response).

Comment (francoisjacques) [reply lib/licensee/projects/github_project.rb:50]: gotcha

**Review by francoisjacques — APPROVED** [review]
LGTM

**Pull request closed without merging**